perf-tool/src/monitor.cpp

```diff
@@ -29,12 +29,15 @@
 #include "agentOptions.hpp"
 #include "infra.hpp"
 #include "json.hpp"
+#include <unordered_map>
 
 //how many times
 using json = nlohmann::json;
 
+std::unordered_map<int, int> m_monitor;
 std::atomic<int> monitorSampleCount{0};
 EventConfig monitorConfig;
+std::mutex monitorMapMutex;
```

Comment:
Maybe move this line close to newly introduced hashtable because they are related. Also, could add a comment that the monitor is for protecting the hashtable